pkg/ddevapp/config.go

```diff
@@ -346,98 +302,98 @@ func (c *Config) RenderComposeYAML() (string, error) {
-	provider, err := c.GetProvider()
+func (app *DdevApp) docrootPrompt() error {
+	provider, err := app.GetProvider()
 	if err != nil {
 		return err
 	}
 
 	// Determine the document root.
-	output.UserOut.Printf("\nThe docroot is the directory from which your site is served. This is a relative path from your application root (%s)", c.AppRoot)
+	output.UserOut.Printf("\nThe docroot is the directory from which your site is served. This is a relative path from your application root (%s)", app.AppRoot)
 	output.UserOut.Println("You may leave this value blank if your site files are in the application root")
 	var docrootPrompt = "Docroot Location"
-	if c.Docroot != "" {
-		docrootPrompt = fmt.Sprintf("%s (%s)", docrootPrompt, c.Docroot)
+	if app.Docroot != "" {
+		docrootPrompt = fmt.Sprintf("%s (%s)", docrootPrompt, app.Docroot)
 	}
 
 	fmt.Print(docrootPrompt + ": ")
-	c.Docroot = util.GetInput(c.Docroot)
+	app.Docroot = util.GetInput(app.Docroot)
 
 	// Ensure the docroot exists. If it doesn't, prompt the user to verify they entered it correctly.
-	fullPath := filepath.Join(c.AppRoot, c.Docroot)
+	fullPath := filepath.Join(app.AppRoot, app.Docroot)
 	if _, err := os.Stat(fullPath); os.IsNotExist(err) {
 		output.UserOut.Errorf("No directory could be found at %s. Please enter a valid docroot\n", fullPath)
-		c.Docroot = ""
-		return c.docrootPrompt()
+		app.Docroot = ""
+		return app.docrootPrompt()
 	}
-	return provider.ValidateField("Docroot", c.Docroot)
+	return provider.ValidateField("Docroot", app.Docroot)
 }
 
 // ConfigExists determines if a ddev config file exists for this application.
-func (c *Config) ConfigExists() bool {
-	if _, err := os.Stat(c.ConfigPath); os.IsNotExist(err) {
+func (app *DdevApp) ConfigExists() bool {
+	if _, err := os.Stat(app.ConfigPath); os.IsNotExist(err) {
 		return false
 	}
 	return true
 }
 
 // appTypePrompt handles the AppType workflow.
-func (c *Config) appTypePrompt() error {
-	provider, err := c.GetProvider()
+func (app *DdevApp) appTypePrompt() error {
+	provider, err := app.GetProvider()
 	if err != nil {
 		return err
 	}
 	var appType string
 	typePrompt := fmt.Sprintf("Application Type [%s]", strings.Join(AllowedAppTypes, ", "))
 
 	// First, see if we can auto detect what kind of site it is so we can set a sane default.
-	absDocroot := filepath.Join(c.AppRoot, c.Docroot)
+	absDocroot := filepath.Join(app.AppRoot, app.Docroot)
 	log.WithFields(log.Fields{
 		"Location": absDocroot,
 	}).Debug("Attempting to auto-determine application type")
 
 	appType, err = DetermineAppType(absDocroot)
 	if err == nil {
 		// If we found an application type just set it and inform the user.
-		util.Success("Found a %s codebase at %s.", appType, filepath.Join(c.AppRoot, c.Docroot))
-		c.AppType = appType
-		return provider.ValidateField("AppType", c.AppType)
+		util.Success("Found a %s codebase at %s.", appType, filepath.Join(app.AppRoot, app.Docroot))
+		app.AppType = appType
```

Comment:
stuttering - app.Type?

Reply:
Changed AppType to Type. It also matches the usage in config.yaml.